SPEC/FILES.md

```diff
@@ -391,6 +391,158 @@ pull(
+```JavaScript
+const validCID = 'QmQ2r6iMNpky5f1m4cnm3Yqw8VSvjuKpTcK1X7dBR1LkJF'
+
+ipfs.files.ls(validCID, function (err, files) {
+  files.forEach((file) => {
+    console.log(file.path)
+  })
+})
+```
+
+A great source of [examples][] can be found in the tests for this API.
+
+#### `lsReadableStream`
+
+> Lists a directory from IPFS that is addressed by a valid IPFS Path. The list will be yielded as Readable Streams.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.lsReadableStream(ipfsPath) -> [Readable Stream][rs]
+
+> **Note:** ipfs.files.ls is currently only for MFS directories. The goal is to converge both functionality.
+
+ipfsPath can be of type:
+
+- [`cid`][cid] of type:
+  - [Buffer][b], the raw Buffer of the cid
+  - String, the base58 encoded version of the cid
+- String, including the ipfs handler, a cid and a path to traverse to, ie:
+  - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66'
+  - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
+  - 'QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
+
+It returns a [Readable Stream][rs] in [Object mode](https://nodejs.org/api/stream.html#stream_object_mode) that will yield objects of the form:
+
+```js
+{
+  depth: 1,
+  name: 'alice.txt',
+  path: 'QmVvjDy7yF7hdnqE8Hrf4MHo5ABDtb5AbX6hWbD3Y42bXP/alice.txt',
+  size: 11696,
+  hash: 'QmZyUEQVuRK3XV7L9Dk26pg6RVSgaYkiSTEdnT2kZZdwoi',
+  type: 'file'
+}
+```
+
+**Example:**
+
+```JavaScript
+const validCID = 'QmQ2r6iMNpky5f1m4cnm3Yqw8VSvjuKpTcK1X7dBR1LkJF'
+
+const stream = ipfs.files.getReadableStream(validCID)
+
+stream.on('data', (file) => {
+  // write the file's path and contents to standard out
+  console.log(file.path)
+})
+```
+
+A great source of [examples][] can be found in the tests for this API.
+
+#### `lsPullStream`
+
+> Fetch a file or an entire directory tree from IPFS that is addressed by a valid IPFS Path. The files will be yielded as Readable Streams.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.lsPullStream(ipfsPath) -> [Pull Stream][ps]
+
+> **Note:** ipfs.files.ls is currently only for MFS directories. The goal is to converge both functionality.
+
+
+ipfsPath can be of type:
+
+- [`cid`][cid] of type:
+  - [Buffer][b], the raw Buffer of the cid
+  - String, the base58 encoded version of the cid
+- String, including the ipfs handler, a cid and a path to traverse to, ie:
+  - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66'
+  - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
+  - 'QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
+
+It returns a [Pull Stream][os] that will yield objects of the form:
+
+```js
+{
+  depth: 1,
+  name: 'alice.txt',
+  path: 'QmVvjDy7yF7hdnqE8Hrf4MHo5ABDtb5AbX6hWbD3Y42bXP/alice.txt',
+  size: 11696,
+  hash: 'QmZyUEQVuRK3XV7L9Dk26pg6RVSgaYkiSTEdnT2kZZdwoi',
+  type: 'file'
+}
+```
+
+**Example:**
+
+```JavaScript
+const validCID = 'QmQ2r6iMNpky5f1m4cnm3Yqw8VSvjuKpTcK1X7dBR1LkJF'
+
+const stream = ipfs.files.getReadableStream(validCID)
```

Comment:
shouldn't this be `ifs.lsPullStream` instead?

Reply:
thanks for the catch! :)